src/Workspaces/CSharp/Portable/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -552,14 +553,11 @@ private static bool AtEndOfIncompleteStringOrCharLiteral(SyntaxToken token, int
                 for (var i = fieldIndex; i < members.Count; i++)
                 {
                     var member = members[i];
-                    if (textSpan.Contains(member.Span))
+                    if (textSpan.Contains(member.Span) ||
+                        (allowPartialSelection && textSpan.OverlapsWith(member.Span)))
```

Comment:
overlaping any part of the span seems liek it's way too large.  what if there is a large initializer and the caret is in it?

Reply:
True.  I'll update it so it has to contain some portion of the identifier name in order to be included.